### PR TITLE
Add def_private_var

### LINKED ISSUE #221: Write a routine for defining internals/private params
When the params are defined in a traditional way, shellcheck checks if the params are used in the current `.sh` file. Very often they are not, so in these cases it's necessary to use the shellcheck directive `shellcheck disable=SC2034`. To make defining internals/private params more neat, write a special routine and name it `def_priv_var`.

### PR DESCRIPTION
**Pull request by Themanwhosmellslikesugar**
Close #221

# Please make sure all below checkboxes in the Mandatory section have been checked before submitting your PR (also don't forget to pay attention to the Depending on Circumstances section)

## Mandatory

- [x] The commit message is an imperative and starts with a capital letter (for example, `Add something`, `Remove something`, `Fix something`, etc.).
- [x] I made sure I hadn't put a period at the end of the the commit message(s).

## Depending on Circumstances

Some of the items in the section become mandatory if you are a first-time contributor and/or your changes are relevant to the items.

- [ ] Adding a new parameter I didn't forget to reflect it in `README.md`.
- [ ] Adding a new parameter or modifying an existing one I didn't break the alphabetical order.
- [ ] Adding a new utility written in Python I didn't forget to reflect it in `pieman/README.md`.
- [ ] Adding a new utility written in Python I didn't forget to increase the version number of the [pieman](https://pypi.org/project/pieman/) package in `pieman/setup.py`, `pieman/pieman/__init__.py` and `essentials.sh`.
- [ ] (for first-time contributors) One of the commits in the pull request adds me to the **Acknowledgements** section in `AUTHORS.md`.

### CODE REVIEW

File: essentials.sh [hunk 1]
```diff
@@ -196,7 +218,7 @@ def_protected_var() {
 def_bool_var() {
     local var=$1
     local val=$2
-    
+
```
Comment (eugulixes) [inline essentials.sh]: This change is redundant for solving the task.

File: pieman.sh [hunk 2]
```diff
@@ -149,8 +149,8 @@ BOOT=${BUILD_DIR}/${PROJECT_NAME}/boot
 
 MOUNT_POINT=${BUILD_DIR}/${PROJECT_NAME}/mount_point
 
-# shellcheck disable=SC2034
-ETC=${R}/etc
+
+def_priv_var ETC "${R}/etc"
```
Comment (eugulixes) [inline pieman.sh]: Do the same thing with other private params.

File: test/test_essentials.sh [hunk 3]
```diff
@@ -25,6 +25,21 @@ setUp() {
 #
 # Let the test begin
 #
+test_defining_private_variables(){
+    priv="test"
+
+    def_priv_var priv "default"
+    >&2 echo "+ "priv"=${priv}"
```
Comment (eugulixes) [inline test/test_essentials.sh]: It's not necessary to echo what the test is doing step by step.

File: essentials.sh [hunk 4]
```diff
@@ -185,6 +185,28 @@ def_protected_var() {
...
+# Arguments:
+#     Variable name
+#     Value by default
+# Returns:
+#     None
+def_priv_var() {
+    local var=$1
+    local val=$2
+
+    if check_if_variable_is_set ${var}; then
```
Comment (eugulixes) [inline essentials.sh]: I guess the check is redundant for private params. The function must be very simple. The main task is wrapping declaring variables -- there's no need to make it do more.

File: pieman.sh [hunk 5]
```diff
@@ -149,42 +149,34 @@ BOOT=${BUILD_DIR}/${PROJECT_NAME}/boot
...
 MOUNT_POINT=${BUILD_DIR}/${PROJECT_NAME}/mount_point
 
-# shellcheck disable=SC2034
-ETC=${R}/etc
 
-# shellcheck disable=SC2034
-USR_BIN=${R}/usr/bin
+def_priv_var ETC "${R}/etc"
+
+def_priv_varUSR_BIN "${R}/usr/bin"
```
Comment (karpa4o4) [inline pieman.sh]: You missed the space between _def_priv_var_ and _USR_BIN_

File: essentials.sh [hunk 6]
```diff
@@ -185,6 +185,23 @@ def_protected_var() {
     >&2 echo "+ ${var_name}=*****"
 }
 
+#Defines variable assigning the specified value to it if the variable does not
```
Comment (eugulixes) [inline essentials.sh]: `Defines a private variable. The function helps avoid using shellcheck disable=SC2034`.

File: pieman.sh [hunk 7]
```diff
@@ -149,42 +149,34 @@ BOOT=${BUILD_DIR}/${PROJECT_NAME}/boot
...
-# shellcheck disable=SC2034
-ETC=${R}/etc
 
-# shellcheck disable=SC2034
-USR_BIN=${R}/usr/bin
+def_priv_var ETC "${R}/etc"
+
+def_priv_var USR_BIN "${R}/usr/bin"
 
 ARTIFACT="${PIEMAN_DIR}/${BUILD_DIR}/${PROJECT_NAME}.mender"
```
Comment (eugulixes) [inline pieman.sh]: Use `def_priv_var`.

File: essentials.sh [hunk 8]
```diff
@@ -185,6 +185,22 @@ def_protected_var() {
...
+# Defines a private variable. The function helps avoid using shellcheck disable=SC2034
+#
+# Globals:
+#     None
+# Arguments:
+#     Variable name
+#     Value by default
+# Returns:
+#     None
+def_priv_var() {
```
Comment (denismosolov) [inline essentials.sh]: Why `def_priv_var`? `def_priv_var` would be much more clear name

File: essentials.sh [hunk 9]
```diff
@@ -185,6 +185,22 @@ def_protected_var() {
     >&2 echo "+ ${var_name}=*****"
 }
 
+# Defines a private variable. The function helps avoid using shellcheck disable=SC2034
+#
```
Comment (karpa4o4) [inline essentials.sh]: This line is redundant.

File: test/test_essentials.sh [hunk 10]
```diff
@@ -25,6 +25,18 @@ setUp() {
 #
 # Let the test begin
 #
+test_defining_private_variables() {
```
Comment (karpa4o4) [inline test/test_essentials.sh]: In the current implementation, before the first test function there is a line of indentation from the comment. Please save this.

**Pull request merged**